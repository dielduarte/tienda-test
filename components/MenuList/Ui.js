import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Small } from '../Text';

export const Menu = styled.nav`
  list-style-type: none;
`;

export const MenuItemText = styled(Small)`
  margin-left: 10px;
  flex: 2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const MenuIcon = styled(FontAwesomeIcon)`
  margin-right: 1px;
`;

export const MenuItem = styled.div`
  padding: 15px 10px;
  display: flex;
  align-items: center;
  color: ${props => props.theme.white};
  transition: all 0.2s linear;

  &:hover {
    cursor: pointer;
    color: ${props => props.theme.lightBlue};
  }
`;
