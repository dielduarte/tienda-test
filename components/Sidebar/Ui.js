import styled from 'styled-components';
import { Small } from '../Text';

export const SidebarContainer = styled.aside`
  width: 320px;
  height: 100vh;
  overflow: scroll;
  background-color: ${props => props.theme.darkGray};
  border-right: 2px solid ${props => props.theme.gray};
  transition: all 0.3s linear;
  position: absolute;
  left: ${props => (props.isOpen ? '0' : '-320px')};
`;

export const SidebarHeader = styled.div`
  background-color: ${props => props.theme.white};
  padding: 15px;
`;

export const SidebarIcon = styled.img`
  margin-right: 10px;
`;

export const Owner = styled(Small)`
  color: ${props => props.theme.gray};
`;
