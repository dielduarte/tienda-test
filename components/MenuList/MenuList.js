import React from 'react';
import Link from 'next/link';
import { arrayOf, objectOf, string } from 'prop-types';

import { Row } from '../Flex';
import { Menu, MenuItem, MenuItemText, MenuIcon } from './Ui';

function renderItem(item, key) {
  return (
    <Link href={item.href} key={key}>
      <MenuItem>
        <Row>
          <MenuIcon icon="ellipsis-v" />
          <MenuIcon icon="bars" />
        </Row>
        <MenuItemText>{item.title}</MenuItemText>
        <MenuIcon icon="angle-right" size="lg" />
      </MenuItem>
    </Link>
  );
}

function MenuList({ items }) {
  return <Menu>{items.map((item, key) => renderItem(item, key))}</Menu>;
}

MenuList.propsType = {
  items: arrayOf(objectOf({ title: string, href: string }))
};

export default MenuList;
