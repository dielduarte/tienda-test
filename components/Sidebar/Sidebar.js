import React from 'react';

import { Regular } from '../Text';
import { Row, Column } from '../Flex';
import Button from '../Button';
import { SidebarContainer, SidebarHeader, Owner, SidebarIcon } from './Ui';

function Sidebar({ storeName, owner, shopLink, children }) {
  return (
    <SidebarContainer>
      <SidebarHeader>
        <Row>
          <SidebarIcon src="http://localhost:3000/static/logo.svg" />
          <Column>
            <Regular>{storeName}</Regular>
            <Owner>{owner}</Owner>
          </Column>
        </Row>
        <Row>
          <Button
            title={shopLink}
            full
            iconName="external-link-alt"
            bordered
            uppercase={false}
            onClick={() => {}}
          />
        </Row>
      </SidebarHeader>
      {children}
    </SidebarContainer>
  );
}

export default Sidebar;
