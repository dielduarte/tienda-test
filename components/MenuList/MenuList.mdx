---
name: MenuList
---

import '../../config/fontAwesome';
import { Playground, PropsTable } from 'docz'
import MenuList from './MenuList';

# MenuList

<Playground style={{ background: '#333' }}>
  <MenuList items={[
      { title: 'Produtos', href: '/products' },
      { title: 'Adicionar um novo produto', href: '/products/new' }
    ]}
  />
</Playground>