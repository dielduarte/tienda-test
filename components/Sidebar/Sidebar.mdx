---
name: Sidebar
---

import '../../config/fontAwesome';
import { Playground } from 'docz'
import Sidebar from './Sidebar';
import MenuList from '../MenuList';

# Sidebar

<Playground>
  <Sidebar storeName="Minha loja" owner="MartÃ­n Palombo" shopLink="teste-brasil.lojavirtualnuvem.com" />
</Playground>