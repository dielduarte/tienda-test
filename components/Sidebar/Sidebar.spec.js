import { shallow } from 'enzyme';
import 'jest-styled-components';

import Sidebar from './Sidebar';
import MenuList from '../MenuList';

describe('Sidebar', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Sidebar
        storeName="Minha loja"
        owner="MartÃ­n Palombo"
        shopLink="link.paraloja.com"
      >
        <MenuList
          items={[
            { title: 'Produtos', href: '/products' },
            { title: 'Adicionar um novo produto', href: '/products/new' }
          ]}
        />
      </Sidebar>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
