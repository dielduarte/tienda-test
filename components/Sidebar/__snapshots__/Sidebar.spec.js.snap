// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<styled.aside>
  <styled.div>
    <styled.div>
      <styled.img
        src="http://localhost:3000/static/logo.svg"
      />
      <styled.div>
        <styled.h3>
          Minha loja
        </styled.h3>
        <Styled(styled.h3)>
          Mart√≠n Palombo
        </Styled(styled.h3)>
      </styled.div>
    </styled.div>
    <styled.div>
      <Button
        bordered={true}
        full={true}
        iconName="external-link-alt"
        onClick={[Function]}
        title="link.paraloja.com"
        type="primary"
        uppercase={false}
      />
    </styled.div>
  </styled.div>
  <MenuList
    items={
      Array [
        Object {
          "href": "/products",
          "title": "Produtos",
        },
        Object {
          "href": "/products/new",
          "title": "Adicionar um novo produto",
        },
      ]
    }
  />
</styled.aside>
`;
